interp/handler.go

```diff
@@ -315,3 +315,19 @@ func DefaultReadDirHandler() ReadDirHandlerFunc {
 		return ioutil.ReadDir(path)
 	}
 }
+
+// StatHandlerFunc is a handler which gets the file stat. the first argument provides directory to use as
+// basedir if name is relative path
+type StatHandlerFunc func(cwd string, name string, lstat bool) (os.FileInfo, error)
```

Comment:
any reason to pass two strings? the handler could just receive one absolute path, like e.g. ReadDirHandlerFunc.

Reply:
I would also invert the boolean and call it `followSymlinks` instead; then the default `false` behavior is to do less work, and the behavior is overall more intuitive.

Reply:
I need to pass the `cwd` argument because the origin function uses the `r.Dir` to build an absolute path in the case when the `name` contains a relative path.

Reply:
That's fine, but please move the cwd stuff to the caller - there's no reason to make every implementer of the handler deal with it. They should just receive the absolute path. We can make the call to absPath in Runner.stat and Runner.lstat.

Reply:
I moved absPath in the parent functions, cwd arg is dropped from the handler func.